modal/image.py

```diff
@@ -1436,9 +1541,6 @@ def from_aws_ecr(
     def from_dockerfile(
         # Filepath to Dockerfile.
         path: Union[str, Path],
-        # modal.Mount with local files to supply as build context for COPY commands.
-        # NOTE: The remote_path of the Mount should match the Dockerfile's WORKDIR.
-        context_mount: Optional[_Mount] = None,
```

Comment:
We can't remove this argument immediately or we'll break user code.

Instead, make the None value here use the new functionality and add a pending deprecation warning if users set it to non-None, in which the error message tells them to stop providing it since we'll automatically detect files for them based on their COPY commands, and contact us if that doesn't work for them (good for us to be able to patch before we remove it).